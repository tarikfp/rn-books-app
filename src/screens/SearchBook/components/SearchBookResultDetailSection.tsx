import { StyleSheet } from 'react-native';
import { Box } from '~components/Box';
import { Spacing } from '~components/Spacing';
import { AppText } from '~components/Text';

type Props = {
  title: string;
  content: string | string[];
};

export const SearchBookResultDetailSection: React.FC<Props> = ({
  content,
  title,
}) => {
  return (
    <>
      <Box
        paddingVertical="sm"
        justifyContent={'center'}
        borderBottomWidth={1 + StyleSheet.hairlineWidth}
        borderColor="border"
      >
        <AppText color="bodyText" variant={'title2'} textAlign={'left'}>
          {title}
        </AppText>

        <Spacing marginTop="s" />

        <AppText variant={'body'} textAlign={'left'} color="bodyText">
          {typeof content === 'object' ? content.join(', ') : content}
        </AppText>
      </Box>
    </>
  );
};
