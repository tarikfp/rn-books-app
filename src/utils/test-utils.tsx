import { ThemeProvider } from '@shopify/restyle';
import { render, RenderOptions } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { theme } from '~theme/core';
import { ThemePreferenceContext } from '~theme/useThemePreference';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ThemePreferenceContext.Provider
      value={{
        isDarkTheme: false,
        toggleTheme: () => {},
      }}
    >
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </ThemePreferenceContext.Provider>
  );
};

const withAllTheProviders = (ui: React.ReactNode) => (
  <AllTheProviders children={ui} />
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderer = (ui: React.ReactElement) =>
  renderer.create(withAllTheProviders(ui));

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { customRender as render };
// override renderer method
export { customRenderer as createRenderer };
