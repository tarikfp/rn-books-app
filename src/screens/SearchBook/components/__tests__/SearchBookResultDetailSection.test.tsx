import React from 'react';
import { createRenderer, render } from '~utils/test-utils';
import { SearchBookResultDetailSection } from '../SearchBookResultDetailSection';

describe('SearchBookResultDetailSection', () => {
  it('renders content with string', () => {
    const tree = createRenderer(
      <SearchBookResultDetailSection title="Title" content="Sample Content" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders content with array', () => {
    const tree = createRenderer(
      <SearchBookResultDetailSection
        title="Title"
        content={['Item 1', 'Item 2', 'Item 3']}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders content as a string', () => {
    const { getByText } = render(
      <SearchBookResultDetailSection title="Title" content="Sample Content" />,
    );

    const title = getByText('Title');
    const content = getByText('Sample Content');

    expect(title).toBeTruthy();
    expect(content).toBeTruthy();
  });

  it('renders content as an array', () => {
    const { getByText } = render(
      <SearchBookResultDetailSection
        title="Title"
        content={['Item 1', 'Item 2', 'Item 3']}
      />,
    );

    const stringArrayText = getByText(
      ['Item 1', 'Item 2', 'Item 3'].join(', '),
    );

    expect(stringArrayText).toBeTruthy();
  });
});
