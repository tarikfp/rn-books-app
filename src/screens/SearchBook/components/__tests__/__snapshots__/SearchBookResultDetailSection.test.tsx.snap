// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBookResultDetailSection renders content with array 1`] = `
<View
  style={
    [
      {
        "borderBottomWidth": 1.5,
        "borderColor": "#E7E8E9",
        "justifyContent": "center",
        "paddingVertical": 12,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#16171C",
          "fontFamily": "Inter-Medium",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": -0.01,
          "lineHeight": 24,
          "textAlign": "left",
        },
      ]
    }
  >
    Title
  </Text>
  <View
    style={
      [
        {
          "marginTop": 8,
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#16171C",
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": -0.01,
          "lineHeight": 20,
          "textAlign": "left",
        },
      ]
    }
  >
    Item 1, Item 2, Item 3
  </Text>
</View>
`;

exports[`SearchBookResultDetailSection renders content with string 1`] = `
<View
  style={
    [
      {
        "borderBottomWidth": 1.5,
        "borderColor": "#E7E8E9",
        "justifyContent": "center",
        "paddingVertical": 12,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#16171C",
          "fontFamily": "Inter-Medium",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": -0.01,
          "lineHeight": 24,
          "textAlign": "left",
        },
      ]
    }
  >
    Title
  </Text>
  <View
    style={
      [
        {
          "marginTop": 8,
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#16171C",
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": -0.01,
          "lineHeight": 20,
          "textAlign": "left",
        },
      ]
    }
  >
    Sample Content
  </Text>
</View>
`;
